python/lance/__init__.py

```diff
@@ -68,15 +76,26 @@ def dataset(
     """
     if not filesystem:
         from pyarrow.fs import _resolve_filesystem_and_path
+
         filesystem, uri = _resolve_filesystem_and_path(uri, filesystem)
 
     if version is None:
-        if (
-            filesystem.get_file_info(os.path.join(uri, "_latest.manifest")).type
-            == pa.fs.FileType.NotFound
-        ):
+        if not _is_versioned(filesystem, uri):
             return _dataset_plain(uri, filesystem=filesystem, **kwargs)
 
+        if asof is not None:
+            ds = dataset(uri, version, filesystem=filesystem, **kwargs)
+            version = get_version_asof(ds, asof)
+
+    return _get_versioned_dataset(filesystem, uri, version)
+
+
+def _is_versioned(filesystem, uri):
```

Comment:
not sure what does this mean. it checks whether latest manifest exist? why it is related to versioned?

Or you mean it is a plain dataset? Should we just rename this function to `is_plain_dataset`?

Reply:
i'm just refactoring what you had in there

Reply:
sure i can call this "is_plain_dataset"

Reply:
done